Modules/_pickle.c

```diff
@@ -3708,6 +3710,15 @@ save_reduce(PicklerObject *self, PyObject *args, PyObject *obj)
         return -1;
     }
 
+    if (state_setter == Py_None)
+        state_setter = NULL;
+    else if (!PyFunction_Check(state_setter)) {
```

Comment:
Ah, I overlooked this. `PyFunction_Check` only accepts pure Python functions. For arbitrary callables, you should use `PyCallable_Check` instead...

Reply:
Ah, did not know about that. Fixed.